lncli: Global flags to accept ENV vars overrides

gcaracuel opened the pull request:
## Change Description
Adds ability to global flags to de config using ENV variables.
This is extremely useful when running `lncli` in containers and lnd not using default values or they do not exist at all.

## Steps to Test
Compile `lncli` run no command or help command manually.
Testing is covered by `urfave/cli` library

## Pull Request Checklist
### Testing
- [x] Your PR passes all CI checks.
- [x] Tests covering the positive and negative (error paths) are included.
- [x] Bug fixes contain tests triggering the bug to prevent regressions.

### Code Style and Documentation
- [x] The change obeys the [Code Documentation and Commenting](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md#CodeDocumentation) guidelines, and lines wrap at 80.
- [x] Commits follow the [Ideal Git Commit Structure](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md#IdealGitCommitStructure).
- [x] Any new logging statements use an appropriate subsystem and logging level.
- [x]  [There is a change description in the release notes](https://github.com/lightningnetwork/lnd/tree/master/docs/release-notes), or `[skip ci]` in the commit message for small changes.

📝 Please see our  [Contribution Guidelines](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md) for further guidance.

guggero commented on cmd/lncli/main.go:
```diff
@@ -342,38 +342,44 @@ func main() {
+			Name:   "rpcserver",
+			Value:  defaultRPCHostPort,
+			Usage:  "The host:port of LN daemon.",
+			EnvVar: "LNCLI_RPCSERVER",
```
Cool, I wasn't even aware that the CLI library supported this feature. Very cool!

gcaracuel replied:
It is, so it is "free" to add, well tested and self-documenting :)

guggero reviewed (approved):
tACK, LGTM :tada:

yyforyongyu commented on cmd/lncli/main.go:
```diff
@@ -342,38 +342,44 @@ func main() {
+			Name:   "rpcserver",
+			Value:  defaultRPCHostPort,
+			Usage:  "The host:port of LN daemon.",
+			EnvVar: "LNCLI_RPCSERVER",
```
make them into constants at the top?

yyforyongyu reviewed (changes requested):
Very nice addition✨ Could you remove the `skip ci` flag so we can run the linter check? Think it failed locally. Also I think this change, though very small, deserves a release note so others know this nice addition!

gcaracuel commented:
> Very nice addition✨ Could you remove the `skip ci` flag so we can run the linter check? Think it failed locally. Also I think this change, though very small, deserves a release note so others know this nice addition!

Added new entry to release note for next version. Removed `[skip ci]` & moved ENV names to constants as per your recommendation. 
Thanks for the feedback.

gcaracuel commented:
Fixed lint issues

guggero commented:
@gcaracuel this PR is still missing the entry to the `docs/release-notes/release-notes-0.17.0.md`.

guggero commented:
Thanks! Can you please squash your commits, then we're good to go.

guggero commented on cmd/lncli/main.go:
```diff
@@ -34,6 +34,18 @@ const (
+	envVarMacaroonTimeout = "LNCLI_MACAROONTIMEOUT"
+	envVarMacaroonIP      = "LNCLI_MACAROONIP"
+	envVarProfile         = "LNCLI_PROFILE"
+	envVarMacFromJAR      = "LNCLI_MACFROMJAR"
```
micro nit: In this context, jar isn't an acronym (not Java ARchive) but the glass container you might put baked cookies in to keep them fresh. So can just be `envVarMacFromJar` ;-)

gcaracuel replied:
oh I will fix that in a minute, I am not 100% familiarised with 100% of the flags. That will be fixed and squashed again in a min.

guggero reviewed (approved):
Thanks, LGTM :tada:

gcaracuel commented:
I did a huge mess trying to solve a stupid merge conflict I think I am gonna need to reopen this

yyforyongyu reviewed (commented):
Should be good to go! But it looks like the latest push removed the entry from the release note?
